core/src/Cuda/Kokkos_Cuda_Parallel_Team.hpp

```diff
@@ -198,12 +198,21 @@ class TeamPolicyInternal<Kokkos::Cuda, Properties...>
   }
 
   inline static int scratch_size_max(int level) {
-    return (
-        level == 0 ? 1024 * 40 :  // 48kB is the max for CUDA, but we need some
-                                  // for team_member.reduce etc.
-            20 * 1024 *
-                1024);  // arbitrarily setting this to 20MB, for a Volta V100
-                        // that would give us about 3.2GB for 2 teams per SM
+    // Cuda Teams use (team_size + 2)*sizeof(double) shared memory for team
+    // reductions. They also use one int64_t in static shared memory for a
+    // shared ID. Furthermore, they use additional scratch memory in some
+    // reduction scenarios, which depend on the size of the value_type and is
+    // NOT captured here.
```

Comment:
Again, please add something in the spirit of this comment as a comment on `Kokkos::TeamPolicy::scratch_size_max` so callers of the public API have an indication that other things may eat into this value.

Reply:
Note that the (admittedly stupid) way this works: this IS the public function, because TeamPolicy inherits from this guy. Turns out there is no scratch_size_max inside the Kokkos_ExecPolicy.hpp ...

Reply:
Oh and the prototype for TeamPolicyInternal in that header is unused, because everything real happens in partial specializations ...

Reply:
that said: working on a PR for the documentation

Reply:
https://github.com/kokkos/kokkos-core-wiki/pull/234